packages/modular-scripts/src/__tests__/esmView.test.ts

```diff
@@ -310,19 +305,97 @@ describe('modular-scripts', () => {
+        await fs.readFile(path.join(baseDir, 'index-CWZPL7BC.js'))
       ).toString();
       expect(
         prettier.format(indexFile, {
           filepath: 'index-F6YQ237K.js',
         }),
       ).toMatchSnapshot();
-      expect(indexFile).toContain(`https://mycustomcdn.net/react?version=`);
+      expect(indexFile).toContain(`https://mycustomcdn.net/react@1`);
+      expect(indexFile).toContain(`https://mycustomcdn.net/lodash@^4.17.21`);
       expect(indexFile).toContain(
-        `https://mycustomcdn.net/lodash?version=^4.17.21`,
+        `https://mycustomcdn.net/lodash.merge@^4.6.2`,
+      );
+    });
+  });
+
+  describe('WHEN building a esm-view with various kinds of package dependencies with webpack', () => {
+    beforeAll(async () => {
+      await fs.copyFile(
+        path.join(__dirname, 'TestViewPackages.test-tsx'),
+        path.join(packagesPath, targetedView, 'src', 'index.tsx'),
+      );
+
+      const packageJsonPath = path.join(
+        packagesPath,
+        targetedView,
+        'package.json',
+      );
+      const packageJson = (await fs.readJSON(
+        packageJsonPath,
+      )) as CoreProperties;
+
+      await fs.writeJSON(
+        packageJsonPath,
+        Object.assign(packageJson, {
+          dependencies: {
+            lodash: '^4.17.21',
+            'lodash.merge': '^4.6.2',
+            'regular-table': '^0.5.6',
+          },
+        }),
+      );
+
+      await execa('yarnpkg', [], {
+        cwd: modularRoot,
+        cleanup: true,
+      });
+
+      await modular('build sample-esm-view', {
+        stdio: 'inherit',
+        env: {
+          EXTERNAL_CDN_TEMPLATE: 'https://mycustomcdn.net/[name]@[version]',
+        },
+      });
+    });
+
+    it('THEN outputs the correct directory structure', () => {
+      expect(tree(path.join(modularRoot, 'dist', 'sample-esm-view')))
+        .toMatchInlineSnapshot(`
+        "sample-esm-view
+        ├─ asset-manifest.json #17u5ev2
+        ├─ index.html #17sfbiz
+        ├─ package.json
+        └─ static
+           └─ js
+              ├─ _trampoline.js #16jj7re
+              ├─ main.55a7c313.js #1gya3xe
+              └─ main.55a7c313.js.map #168abd0"
+      `);
+    });
+
+    it('THEN rewrites the dependencies', async () => {
+      const baseDir = path.join(
+        modularRoot,
+        'dist',
+        'sample-esm-view',
+        'static',
+        'js',
       );
+
+      const indexFile = (
+        await fs.readFile(path.join(baseDir, 'main.55a7c313.js'))
+      ).toString();
+      expect(
+        prettier.format(indexFile, {
+          filepath: 'index-F6YQ237K.js',
+        }),
+      ).toMatchSnapshot();
+      expect(indexFile).toContain(`https://mycustomcdn.net/react@1`);
+      expect(indexFile).toContain(`https://mycustomcdn.net/lodash@^4.17.21`);
       expect(indexFile).toContain(
-        `https://mycustomcdn.net/lodash.merge?version=^4.6.2`,
+        `https://mycustomcdn.net/lodash.merge@^4.6.2`,
```

Comment:
Should we be using hats or exact versions for consistency at runtime?

Reply:
Hats, we already have a test for exact versions (resolutions)